tools/shared/src/main/scala/org/scalajs/core/tools/linker/analyzer/Analyzer.scala

```diff
@@ -587,10 +607,21 @@ private final class Analyzer(config: CommonPhaseConfig,
 
         if (isScalaClass) {
           accessData()
-          ancestors.foreach(_.subclassInstantiated())
 
-          for ((methodName, from) <- delayedCalls)
-            callMethodResolved(methodName)(from)
+          val allMethodsCalledLogs = for (ancestor <- ancestors) yield {
+            ancestor.subclassInstantiated()
+            ancestor.instantiatedSubclasses ::= this
+            ancestor.methodsCalledLog
+          }
+
+          for {
+            log <- allMethodsCalledLogs
+            logEntry <- log
+          } {
+            val methodName = logEntry._1
+            implicit val from = logEntry._2
```

Comment:
Not necessarily, but it's definitely more efficient. Destructuring directly in the for loop creates an awful `withFilter` and other nasty inefficient things. We carefully avoid that kind of caveats in all the algorithms that are not incremental in the linker.